snuba/migrations/policies.py

```diff
@@ -0,0 +1,75 @@
+from abc import ABC, abstractmethod
+
+from snuba.migrations.groups import get_group_loader
+from snuba.migrations.runner import MigrationKey, Runner
+from snuba.migrations.status import Status
+
+
+class MigrationPolicy(ABC):
+    """
+    A MigrationPolicy implements an `is_allowed` method
+    that determines whether or not a migration can be applied
+    from snuba admin.
+
+    A policy can be used by a migration group to determine the
+    level of access a group has to run migrations.
+    """
+
+    @abstractmethod
+    def can_run(self, migration_key: MigrationKey) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    def can_reverse(self, migration_key: MigrationKey) -> bool:
+        raise NotImplementedError
+
+
+class ReadOnlyPolicy(MigrationPolicy):
+    """
+    No migration is allowed to be run or reversed.
+    """
+
+    def can_run(self, migration_key: MigrationKey) -> bool:
+        return False
+
+    def can_reverse(self, migration_key: MigrationKey) -> bool:
+        return False
+
+
+class WriteSafeAndPendingPolicy(MigrationPolicy):
```

Comment:
but this isn't just running non-blocking migrations, it's also reversing migrations, which would then be considered blocking. I think because blocking is used to describe a whole migration vs the migration and the action it's confusing.

Reply:
Yes, i suspect there is a better option than this. But `WriteSafeAndPendingPolicy` is confusing for me and it also doesn't capture all of these caveats that you mentioned.

Reply:
The only thing I can think of would be like `RunNonBlockingReverseBlockingPolicy` which is kind of long (although `WriteSafeAndPendingPolicy` is already long)

Reply:
Imo `RunNonBlocking` is both shorter and better captures the intent of the policy which is only to run the non blocking migrations. The ability to reverse the pending migration is only really in service of being able to revert what you just did via the tool in case it failed halfway. IMO this policy should actually only allow pending, non-blocking migrations to be reversed as that's all the tool allowed in the first place.